erpnext/regional/report/uae_vat_201/uae_vat_201.py

```diff
@@ -0,0 +1,336 @@
+		},
+		{
+			"fieldname": "legend",
+			"label": _("Legend"),
+			"fieldtype": "Data",
+			"width": 300
+		},
+		{
+			"fieldname": "amount",
+			"label": _("Amount (AED)"),
+			"fieldtype": "Currency",
+			"width": 125,
+			"options": "currency"
+		},
+		{
+			"fieldname": "vat_amount",
+			"label": _("VAT Amount (AED)"),
+			"fieldtype": "Currency",
+			"width": 150,
+			"options": "currency"
+		}
+	]
+
+def get_data(filters = None):
+	"""Returns the list of dictionaries. Each dictionary is a row in the datatable and chart data."""
+	data = []
+	emirates, amounts_by_emirate = append_vat_on_sales(data, filters)
+	append_vat_on_expenses(data, filters)
+	return data, emirates, amounts_by_emirate
+
+def get_chart(emirates, amounts_by_emirate):
+	"""Returns chart data."""
+	labels = []
+	amount = []
+	vat_amount = []
+	for d in emirates:
+		if d in amounts_by_emirate:
+			amount.append(amounts_by_emirate[d]["raw_amount"])
+			vat_amount.append(amounts_by_emirate[d]["raw_vat_amount"])
+			labels.append(d)
+
+	datasets = []
+	datasets.append({'name': _('Amount (AED)'), 'values':  amount})
+	datasets.append({'name': _('Vat Amount (AED)'), 'values': vat_amount})
+
+	chart = {
+			"data": {
+				'labels': labels,
+				'datasets': datasets
+			}
+		}
+
+	chart["type"] = "bar"
+	chart["fieldtype"] = "Currency"
+	return chart
+
+def append_vat_on_sales(data, filters):
+	"""Appends Sales and All Other Outputs."""
+	append_data(data, '', _('VAT on Sales and All Other Outputs'), '', '')
+
+	emirates, amounts_by_emirate = standard_rated_expenses_emiratewise(data, filters)
+
+	append_data(data, '2', _('Tax Refunds provided to Tourists under the Tax Refunds for Tourists Scheme'),
+		frappe.format((-1) * get_tourist_tax_return_total(filters), 'Currency'),
+		frappe.format((-1) * get_tourist_tax_return_tax(filters), 'Currency'))
+
+	append_data(data, '3', _('Supplies subject to the reverse charge provision'),
+		frappe.format(get_reverse_charge_total(filters), 'Currency'),
+		frappe.format(get_reverse_charge_tax(filters), 'Currency'))
+
+	append_data(data, '4', _('Zero Rated'), 
+		frappe.format(get_zero_rated_total(filters), 'Currency'), "-")
+
+	append_data(data, '5', _('Exempt Supplies'),
+		frappe.format(get_exempt_total(filters), 'Currency'),"-")
+
+	append_data(data, '', '', '', '')
+	
+	return emirates, amounts_by_emirate
+
+def standard_rated_expenses_emiratewise(data, filters):
+	"""Append emiratewise standard rated expenses and vat."""
+	total_emiratewise = get_total_emiratewise(filters)
+	emirates = get_emirates()
+	amounts_by_emirate = {}
+	for d in total_emiratewise:
+		emirate, amount, vat= d
+		amounts_by_emirate[emirate] = {
+				"legend": emirate,
+				"raw_amount": amount,
+				"raw_vat_amount": vat,
+				"amount": frappe.format(amount, 'Currency'),
+				"vat_amount": frappe.format(vat, 'Currency'),
+			}
+	amounts_by_emirate = append_emiratewise_expenses(data, emirates, amounts_by_emirate)
+
+	for d, emirate in enumerate(emirates, 97):
```

Comment:
- Why are you setting the `start` to `97`?
- Need better variable naming

Reply:
I am starting enumerate with 97 because that is the ASCII for 'a'
The rows in the report are numbered a (1a, 1b, 1c ...)